File: Competive Programming/Java/HackerRank/minimum-absolute-difference-in-an-array.java
import java.util.Arrays;
import java.util.Scanner;


public class Solution {
    static int minimumAbsoluteDifference(int[] arr, int length) {
        Arrays.sort(arr);

        int minimumDifference = Math.abs(arr[0] - arr[1]);

        for (int i=0; i < length - 1; ++i) {
            int diff = Math.abs(arr[i] - arr[i+1]);
            if (diff < minimumDifference)
                minimumDifference = diff;
        }

        return minimumDifference;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int length = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
        int[] arr = new int[length];
        for (int i = 0; i < length; i++) {
            arr[i] = scanner.nextInt();
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
        }

        System.out.println(minimumAbsoluteDifference(arr, length));
        scanner.close();
    }
}

File: Competive Programming/Python/HackerRank/minimum-absolute-difference-in-an-array.py
def minimumAbsoluteDifference(arr, n):
    min_diff = abs(arr[0] - arr[1])

    for i in range(1, n - 1):
        diff = abs(arr[i] - arr[i+1])
        if diff < min_diff:
            min_diff = diff
    return min_diff


if __name__ == '__main__':
    n = int(input())
    arr = sorted(map(int, input().split()))

    print(minimumAbsoluteDifference(arr, n))

